packages/teams-js/src/public/marketplace.ts

```diff
@@ -0,0 +1,361 @@
+     */
+    AdminUser = 'AdminUser',
+    /**
+     * @hidden
+     * The cart is created by end user of an organization.
+     */
+    EndUser = 'EndUser',
+  }
+
+  /**
+   * @hidden
+   * Represents the status of the cart.
+   * @beta
+   */
+  export enum CartStatus {
+    /**
+     * @hidden
+     * Cart is created but not checked out yet.
+     */
+    Open = 'Open',
+    /**
+     * @hidden
+     * Cart is checked out but not completed yet.
+     */
+    Processing = 'Processing',
+    /**
+     * @hidden
+     * Indicate checking out is completed and the host should
+     * return a new cart in the next getCart call.
+     */
+    Processed = 'Processed',
+    /**
+     * @hidden
+     * Indicate checking out process is manually cancelled by the user
+     */
+    Closed = 'Closed',
+    /**
+     * @hidden
+     * Indicate checking out is failed and the host should
+     * return a new cart in the next getCart call.
+     */
+    Error = 'Error',
+  }
+  /**
+   * @hidden
+   * Represents the parameters to update the cart items.
+   * @beta
+   */
+  export interface AddOrUpdateCartItemsParams {
+    /**
+     * @hidden
+     * The uuid of the cart to be updated, target on the cart
+     * being checked out  if cartId is not provided.
+     */
+    cartId?: string;
+    /**
+     * @hidden
+     * A list of cart items object, for each item,
+     * if item id exists in the cart, overwrite the item price and quantity,
+     * otherwise add new items to cart.
+     */
+    cartItems: CartItem[];
+  }
+  /**
+   * @hidden
+   * Represents the parameters to remove the cart items.
+   * @beta
+   */
+  export interface RemoveCartItemsParams {
+    /**
+     * @hidden
+     * The uuid of the cart to be updated, target on the cart
+     * being checked out if cartId is not provided.
+     */
+    cartId?: string;
+    /**
+     * @hidden
+     * A list of cart id, delete the cart item accordingly.
+     */
+    cartItemIds: string[];
+  }
+  /**
+   * @hidden
+   * Represents the parameters to update the cart status.
+   * @beta
+   */
+  export interface UpdateCartStatusParams {
+    /**
+     * @hidden
+     * The uuid of the cart to be updated, target on the cart
+     * being checked out if cartId is not provided.
+     */
+    cartId?: string;
+    /**
+     * @hidden
+     * Status of the cart.
+     */
+    cartStatus: CartStatus;
+    /**
+     * @hidden
+     * Extra info to the status.
+     */
+    statusInfo?: string;
+  }
+
+  /**
+   * @hidden
+   * Get the cart object owned by the host to checkout.
+   * @returns A promise of the cart object.
+   * @beta
+   */
+  export function getCart(): Promise<Cart> {
+    return new Promise<Cart>((resolve) => {
+      ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+      if (!isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+      resolve(sendAndHandleSdkError('marketplace.getCart'));
+    });
```

Comment:
Is this "return new Promise" pattern necessary? I see it's used multiple places in this file. It seems like it adds an extra unnecessary layer of promise wrapping/etc. Could this just be written more straightforwardly as:
```
      ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
      if (!isSupported()) {
        throw errorNotSupportedOnPlatform;
      }
      return sendAndHandleSdkError('marketplace.getCart');
    });
```
?